llvm/include/llvm/Passes/StandardInstrumentations.h

```diff
@@ -578,6 +579,84 @@ class PrintCrashIRInstrumentation {
   static void SignalHandler(void *);
 };
 
+/// A class to collect and print dropped debug information variable statistics.
+/// After every LLVM IR pass is run, it will print how many #dbg_values were
+/// dropped due to that pass.
+class DroppedVariableStats {
+public:
+  DroppedVariableStats(bool DroppedVarStatsEnabled) {
+    if (DroppedVarStatsEnabled)
+      llvm::outs()
+          << "Pass Level, Pass Name, Num of Dropped Variables, Func or "
+             "Module Name\n";
+  };
+  // We intend this to be unique per-compilation, thus no copies.
+  DroppedVariableStats(const DroppedVariableStats &) = delete;
+  void operator=(const DroppedVariableStats &) = delete;
+
+  void registerCallbacks(PassInstrumentationCallbacks &PIC);
+  void runBeforePass(StringRef PassID, Any IR);
+  void runAfterPass(StringRef PassID, Any IR, const PreservedAnalyses &PA);
+  void runAfterPassInvalidated(StringRef PassID, const PreservedAnalyses &PA);
+  bool getPassDroppedVariables() { return PassDroppedVariables; }
+
+private:
+  bool PassDroppedVariables = false;
+  /// VarID is a unique key that represents a #dbg_value
+  using VarID =
+      std::tuple<const DIScope *, const DIScope *, const DILocalVariable *>;
+
+  struct DebugVariables {
+    /// Maps the name of an llvm::Function to a DenseSet of VarIDs before an
+    /// optimization pass has run.
+    DenseMap<const Function *, DenseSet<VarID>> DebugVariablesBefore;
+    /// Maps the name of an llvm::Function to a DenseSet of VarIDs after an
+    /// optimization pass has run.
+    DenseMap<const Function *, DenseSet<VarID>> DebugVariablesAfter;
```

Comment:
And why are these two maps? I would expect them to have exactly identical keys, too?
i.e.: 

struct DebugVariables {
    DenseSet<VarID> DebugVariablesBefore;
    DenseSet<VarID> DebugVariablesAfter;
};

SmallVector<DenseMap<Function*, DebugVariables>> DebugVariablesStack;